test/navigation/NavigatorSpec.scala

```diff
@@ -43,6 +43,85 @@ class NavigatorSpec extends SpecBase with ScalaCheckPropertyChecks with Generato
               .mustBe(routes.IndexController.onPageLoad())
         }
       }
+
+      "must go from 'Which categories of hallmarks are relevant to this arrangement' page " +
+        "to 'Which parts of hallmark A apply to this arrangement?' page " +
+        "when checkbox A is selected" in {
+
+        forAll(arbitrary[UserAnswers]) {
+          answers =>
+
+            val updatedAnswers =
+              answers.set(HallmarkCategoriesPage, HallmarkCategories.enumerable.withName("categoryA").toSet)
+                  .success
+                  .value
+
+            navigator
+              .nextPage(HallmarkCategoriesPage, NormalMode, updatedAnswers)
+              .mustBe(routes.HallmarkAController.onPageLoad(NormalMode))
+        }
+      }
+
+      "must go from 'Which parts of hallmark A apply to this arrangement?' page " +
+        "to 'Does the arrangement meet the Main Benefit Test?' page " +
+        "when checkbox A1, A2a, A2b & A3 is selected" in {
+
+        forAll(arbitrary[UserAnswers]) {
+          answers =>
+
+            val updatedAnswers =
+              answers.set(HallmarkAPage, HallmarkA.enumerable.withName("A1").toSet)
+                .success
+                .value
+              answers.set(HallmarkAPage, HallmarkA.enumerable.withName("A2a").toSet)
+                .success
+                .value
+              answers.set(HallmarkAPage, HallmarkA.enumerable.withName("A2b").toSet)
+                .success
+                .value
+              answers.set(HallmarkAPage, HallmarkA.enumerable.withName("A3").toSet)
+                .success
+                .value
```

Comment:
this isn't adding each hallmark. You probably want HallmarkA.values.toSet or construct a set from the case objects you need

Reply:
Changed to HallmarkA.values.toSet as advised :) Thanks!